### PR TITLE
Read scala default values, if no fieldDefaultValue annotation is found

### LINKED ISSUE #552: Get default value from case class default paramters
It's very convenient to define default value in case class, and zio-json do so.

### PR DESCRIPTION
**Pull request by 987Nabil**
fixes #552
/claim #552

### CODE REVIEW

**Comment by liewhite** [comment]
Any progress?

**Review by jdegoes — DISMISSED** [review]

**Comment by 987Nabil** [comment]
@jdegoes I fixed the linting errors

**Comment by liewhite** [comment]
I test this locally, it failed on generic field.

**Comment by 987Nabil** [comment]
@liewhite Can you provide a reproducer and tell me for which Scala version?

File: zio-schema-derivation/shared/src/main/scala-3/zio/schema/DeriveSchema.scala [hunk 1]
```diff
@@ -275,13 +275,40 @@ private case class DeriveSchema()(using val ctx: Quotes) extends ReflectionUtils
...
+            .companionClass
+            .declaredMethod(s"$$lessinit$$greater$$default$$$i")
+            .headOption
+            .orElse(
+              from
+                .companionClass
+                .declaredMethod(s"$$apply$$default$$$i")
+                .headOption
+            )
+            .map(Select(Ref(from.companionModule), _).asExpr)
```
Comment (liewhite) [inline zio-schema-derivation/shared/src/main/scala-3/zio/schema/DeriveSchema.scala:291]: Compiling failed on Generic case class here.

**Comment by liewhite** [comment]
> @liewhite Can you provide a reproducer and tell me for which Scala version?

```scala
case class A[T](a: Option[T], b: Int) derives Schema
```
Compilation failed with `not a Expr` error. 

I comment in code view

**Comment by 987Nabil** [comment]
Thanks. I'll take a look soon.

**Comment by liewhite** [comment]
> Thanks. I'll take a look soon.
```scala
if(i.isExpr) {
        i.asExpr
      }else {
        i.appliedToType(TypeRepr.of[Any]).asExpr
      }
```
This works.
But is it safe to use `any` after compiler type check?

**Comment by 987Nabil** [comment]
@liewhite Finally found some time. Your solution is ok, since when we call `toExpr` it ends up being an `Expr[Any]` anyway.
But there have been more problems I needed to fix. Now it looks ok.

@jdegoes I could not use a generic implicit for the test of the generic class. If I do it, the gen macro for Scala 3 fails. But this seems to be a problem that existed before my changes. I'll open a bg ticket.

**Review by vigoo — DISMISSED** [review]

**Review by vigoo — APPROVED** [review]

**Pull request merged**